testplan/testing/multitest/suite.py

```diff
@@ -805,16 +806,27 @@ def testcase_with_pre(*args, **kwargs):
     return testcase_with_pre
 
 
-def run_epilogues(epilogues, name, *args, **kwargs):
+def _run_epilogues(epilogues, name, raise_exc=False, *args, **kwargs):
```

Comment:
This is a bit too many if-else. How about we let _run_epilogues 1) log the exception 2) return the exceptions if any? Then the caller can decide if re-raise?

Reply:
updated. `_run_epilogues` returns all possible exception during running post-testcase functions. if the testcase itself does not raise, then the caller decides to raise the first exception found in epilogues.